lib/pbench/test/unit/agent/task/test_results_push.py

```diff
@@ -224,7 +223,15 @@ def test_push_status(status_code, message, exit_code):
         )
         assert result.exit_code == exit_code, result.stderr
         assert result.stdout == ""
-        assert result.stderr.strip() == "" if not message else message["message"]
+
+        try:
+            err_msg = "" if not message else message["message"]
+        except TypeError:
+            err_msg = message
```

Comment:
Since the value of `message` is determined by the `parametrize()` list which we control, having a `try` block here seems overdone.

Reply:
I opened issue #3374 to deal with the remaining problems and objections. I think I have captured them all.